climpyrical/cmd/preprocess_model.py

```diff
@@ -0,0 +1,225 @@
+            "Path to intput CanRCM4 file "
+            "Note: must have lat, lon"
+            "rlat, rlon and a data variable."
+        ),
+    )
+    parser.add_argument(
+        "-o",
+        "--output_file",
+        help=(
+            "Path to store results."
+            "Caution! Overwrites existing"
+            "file in the directory."
+        ),
+    )
+    parser.add_argument(
+        "-m",
+        "--fill_glaciers",
+        default=True,
+        help=("Boolean, removes spurrious glacier" "points within model"),
+    )
+
+    args = parser.parse_args()
+
+    if args.fill_glaciers is not None:
+        if args.fill_glaciers:
+            fill_glaciers = True
+        else:
+            fill_glaciers = False
+
+    if not args.input_file.endswith(".nc") or not args.output_file.endswith(
+        ".nc"
+    ):
+        raise IOError("Please provide a .nc file.")
+
+    run_processing(args.input_file, args.output_file, fill_glaciers)
+
+
+def run_processing(IN_PATH, OUT_PATH, fill_glacieres):
+    """Completes the preprocessing of the model required
+    for the NRC project.
+    Args:
+        IN_PATH, OUT_PATH (strings): directories of NetCDF4 file
+            input and output. Must give filename, too with extension
+            .nc. Overwites files with same name in same directory.
+        fill_glaciers (bool): whether to fill spurious glacier
+            points with preprocessed mask. Default is True.
+    Returns:
+        Creates a NetCDF4 file at OUT_PATH at target resolution
+            for
+    """
+
+    ds = read_data(IN_PATH)
+    dv = list(ds.data_vars)[0]
+
+    # create handy 2D grids
+    rlon, rlat = np.meshgrid(ds.rlon, ds.rlat)
+    # extract data field
+    mean = ds[dv].values
+
+    # Detect unit conversions
+    temp_dv_names = ["twb", "tas", "heating_degree_days_per_time_period"]
+    if dv in temp_dv_names:
+        print("Temperature field detected. Converting to Kelvin.")
+        mean = ds[dv].values + 273.15
+    else:
+        mean = ds[dv].values
+
+    # import path masks
+    # surface to land mask
+    PATH_MASK = resource_filename(
+        "climpyrical", "nrc_data/land_mask_CanRCM4_sftlf.nc"
+    )
+
+    # glacier point mask developed from SL50
+    PATH_GLACIER_MASK = resource_filename(
+        "climpyrical", "nrc_data/glacier_mask.nc"
+    )
+
+    # load mask data
+    mask = read_data(PATH_MASK)
+
+    # regrid dataset to target resolution
+    mask = regrid_ensemble(mask, "sftlf", 10, copy=True)
+
+    # convert to boolean mask
+    mask = mask["sftlf"] >= 1.0
+
+    # keep original mask, convert to boolean
+    mask_og = read_data(PATH_MASK)["sftlf"].values != 0.0
+    # load and convert glacier mask to boolean
+    glaciermask = read_data(PATH_GLACIER_MASK)["mask"].values != 0.0
+
+    # insert NaN values into glacier points to fill
+    fill_glaciers = True
+    if fill_glaciers:
+        print("Filling spurious glacier points.")
+        mean[glaciermask] = np.nan
```

Comment:
We pass in `fill_glaciers` as a param into this method and then set it to `true` and immediately check the truthiness? This could be testing code that got left in accidentally but if it is not `fill_glaciers` does not need to be a param.

Reply:
A mistake. Good eye. That would have been annoying to troubleshoot.